### PR TITLE
[FEATURE][ML] Client-side "Start" API method for DataFrame analytics config.

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR implements "Start" method for DataFrame analytics configs in high-level REST client.

### CODE REVIEW

**Comment by przemekwitek** [comment]
run elasticsearch-ci/1

File: client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java [hunk 1]
```diff
@@ -1351,6 +1297,42 @@ public void testGetDataFrameAnalyticsConfig_ConfigNotFound() {
...
+            machineLearningClient::putDataFrameAnalytics, machineLearningClient::putDataFrameAnalyticsAsync);
+
+        AcknowledgedResponse startDataFrameAnalyticsResponse = execute(
+            new StartDataFrameAnalyticsRequest(configId),
+            machineLearningClient::startDataFrameAnalytics, machineLearningClient::startDataFrameAnalyticsAsync);
+        assertTrue(startDataFrameAnalyticsResponse.isAcknowledged());
+
+        // TODO(przemyslaw.witek): Instead of sleeping, poll for task state "STOPPED" once GetDataFrameAnalyticsStats API method is
+        //                         implemented in MachineLearningClient
+        Thread.sleep(10 * 1000);
```
Comment (przemekwitek) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java]: Apparently it breaks connections in unrelated tests. 
I'll solve it this way: I'll wait for dest index creation with timeout. When GetStats is implemented, I'll change the implementation to wait for STOPPED.

**Comment by davidkyle** [comment]
Tests in MachineLearningIT are failing with:
```
Cannot delete data frame analytics [test-config] while its status is [analyzing]
```

I'm guessing this is the cleanup code failing to delete a started job. What do you think of pulling the test out of this PR and putting it back in when stop and get stats is implemented. Saves the horrible thread sleep hack too.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java (hunk 1 above)
Comment (przemekwitek) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java]: As suggested by @davidkyle, I'll remove the test for now as currently there is no easy way to make it correct.

**Comment by przemekwitek** [comment]
Totally agree. The "sleep" hack didn't work anyway, so I was looking for other methods to make the test pass. But of course it will be much easier after there is "stop".

**Review by davidkyle — APPROVED** [review]
LGTM. There is a minor change to the java docs to make please

File: client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java [hunk 2]
```diff
@@ -1966,6 +1967,45 @@ public void getDataFrameAnalyticsAsync(GetDataFrameAnalyticsRequest request, Req
             Collections.emptySet());
     }
 
+    /**
+     * Starts Data Frame Analytics
+     * <p>
+     * For additional info
+     * see <a href="https://www.TODO.com">POST Data Frame Analytics documentation</a>
```
[the remark this answers is not in the document]
Comment (przemekwitek) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java]: Done.

**Comment by przemekwitek** [comment]
run elasticsearch-ci/bwc
run elasticsearch-ci/default-distro
run elasticsearch-ci/packaging-sample

**Pull request merged**